Source/ResponseSerialization.swift

```diff
@@ -575,11 +586,11 @@ public final class DecodableResponseSerializer<T: Decodable>: ResponseSerializer
                 throw AFError.responseSerializationFailed(reason: .inputDataNilOrZeroLength)
             }
 
-            guard let emptyValue = Empty.value as? T else {
+            guard let emptyResponseType = T.self as? EmptyResponse.Type else {
                 throw AFError.responseSerializationFailed(reason: .invalidEmptyResponse(type: "\(T.self)"))
             }
 
-            return emptyValue
+            return emptyResponseType.emptyResponseValue() as! T
```

Comment:
This can again be combined into the guard.

Reply:
addressed in 4664bb578a70c3f8a1d3f74082b352d5eea8fd06